pymatgen/transformations/advanced_transformations.py

```diff
@@ -1360,3 +1361,295 @@ def inverse(self):
+                    length = np.linalg.norm(a - projb_a)
+                else:
+                    length = np.linalg.norm(a - projc_a)
+                width = math.sqrt(np.linalg.norm(b) ** 2 - np.linalg.norm(projb_a) ** 2)
+                ab_normal = np.cross(a, b)  # get normal direction from AB plane
+                height = np.linalg.norm(_proj(c, ab_normal))  # project c onto AB plane normal
+
+                self.smallest_dim = min([length, width, height])
+
+                # Get number of atoms
+                superstructure = SupercellTransformation(self.trans_mat).apply_transformation(structure)
+                num_at = superstructure.num_sites
+
+                # Check if constraints are satisfied
+                if self.smallest_dim >= hard_sc_size_threshold \
+                        and num_at >= self.min_atoms and num_at <= self.max_atoms:
+                    return superstructure
+                else:
+                    # Increase threshold until proposed supercell meets requirements
+                    target_sc_size += 0.1
+                    if num_at > self.max_atoms:
+                        raise AttributeError('While trying to solve for the supercell, the max'
+                                             'number of atoms was exceeded. Try lowering the number'
+                                             'of nearest neighbor distances.')
+
+    @property
+    def inverse(self):
+        return None
+
+    @property
+    def is_one_to_many(self):
+        return False
+
+class PerturbSitesTransformation(AbstractTransformation):
+    """
+    Uses a structure, a min/max displacement, a number of displacement
+    values, and a number of supercells per unique displacement value
+    to generate a list of randomly perturbed supercells from a given
+    supercell.
+    """
+
+    def __init__(self,
+                 max_displacement_val,
+                 min_displacement_val,
+                 num_displacements,
+                 structures_per_displacement_val,
+                 floor_displacement=None):
+        """
+        Args:
+            original_structure (Structure): structure to displace, typically
+                already relaxed
+            max_displacement_val (float): maximum displacement value for
+                perturbing the structure (Angstroms)
+            min_displacement_val (float): minimum displacement value for
+                perturbing the structure (Angstroms)
+            num_displacements (int): number of unique displacement values to
+                try, uniformly distributed between 'min_displacement' and
+                'max_displacement'. This argument is ignored if min_displacement
+                is not passed as an argument.
+            structures_per_displacement_val (int): number of perturbed structures to
+                generate for each unique displacement value.
+
+            floor_displacement (Optional float): If None (default), then for a
```

Comment:
Why is this a float? Should this just be a boolean? This docstring makes it sound like the value is not actually used.